src/main/java/com/example/demo1/HelloApplication.java

```diff
@@ -0,0 +1,298 @@
+                    break;
+                case D:
+                    group.translateXProperty().set(group.getTranslateX() + 1);
+                    break;
+                case A:
+                    group.translateXProperty().set(group.getTranslateX() - 1);
+                    break;
+                case UP:
+                    group.rotateByX(10);
+                    break;
+                case DOWN:
+                    group.rotateByX(-10);
+                    break;
+                case RIGHT:
+                    group.rotateByY(10);
+                    break;
+                case LEFT:
+                    group.rotateByY(-10);
+                    break;
+            }
+        });
+
+        primaryStage.setScene(scene);
+    }
+
+    public MeshView getMeshView() {
+        TriangleMesh mesh = new TriangleMesh();
+        mesh.getPoints().addAll(vertices);
+        mesh.getTexCoords().addAll(texture);
+        mesh.getFaces().addAll(faces);
+
+        MeshView meshView = new MeshView();
+        meshView.setMesh(mesh);
+        meshView.setMaterial(new PhongMaterial(Color.YELLOW));
+        meshView.setCullFace(CullFace.NONE);
+
+        return meshView;
+    }
+
+    private void initMouseControl(SmartGroup group, Scene scene, Stage stage) {
+        Rotate xRotate;
+        Rotate yRotate;
+        group.getTransforms().addAll(
+                xRotate = new Rotate(0, Rotate.X_AXIS),
+                yRotate = new Rotate(0, Rotate.Y_AXIS)
+        );
+        xRotate.angleProperty().bind(angleX);
+        yRotate.angleProperty().bind(angleY);
+
+        scene.setOnMousePressed(event -> {
+            anchorX = event.getSceneX();
+            anchorY = event.getSceneY();
+            anchorAngleX = angleX.get();
+            anchorAngleY = angleY.get();
+        });
+
+        scene.setOnMouseDragged(event -> {
+            angleX.set(anchorAngleX - (anchorY - event.getSceneY()));
+            angleY.set(anchorAngleY + anchorX - event.getSceneX());
+        });
+
+        stage.addEventHandler(ScrollEvent.SCROLL, event -> {
+            double delta = event.getDeltaY();
+            group.translateZProperty().set(group.getTranslateZ() + delta);
+        });
+    }
+
+    private void loadVerticesFromFile() {
+        FileChooser fileChooser = new FileChooser();
+        fileChooser.setTitle("Open Vertices File");
+        File file = fileChooser.showOpenDialog(null);
+
+        if (file != null) {
+            try (Scanner scanner = new Scanner(file)) {
+                List<Float> vertexList = new ArrayList<>();
+                while (scanner.hasNextLine()) {
+                    String line = scanner.nextLine().trim();
+                    String[] values = line.split("\\s+");
+
+                    for (int i = 1; i < values.length; i++) {
+                        try {
+                            vertexList.add(Float.parseFloat(values[i]));
```

Comment:
Scanner has methods for reading numbers, nextInt(), nextDouble() as I remember. Maybe they will be more convenient instead of splitting a Spting